src/coreclr/debug/daccess/request.cpp

```diff
@@ -1227,7 +1227,7 @@ ClrDataAccess::GetMethodDescTransparencyData(CLRDATA_ADDRESS methodDesc, struct
 
     SOSDacEnter();
 
-    MethodDesc *pMD = PTR_MethodDesc(TO_TADDR(methodDesc));
+    PTR_MethodDesc pMD = PTR_MethodDesc(TO_TADDR(methodDesc));
```

Comment:
So mostly prevent AV by having the smart pointers do some checks themselves? Or mostly just to skip the instantiation of the object on the host side?

Reply:
Yes, this is to prevent actual AVs/segfaults on Linux/MacOS that are not caught and crash lldb, etc. (they are catch on Windows). This is to make the eestack/dumpstack commands more robust.